classes/01class/exercises/c01-git04/igorjosesantos/README.md

```diff
@@ -0,0 +1,47 @@
+# c01-git04
+
+## Questions (answer inline)
+
+1. Let's suppose you remotely pushed the `my_env.txt` file above. A colleague asks you to remove this information from Git. What's your colleague worried about?
+
+> He's worried about the sensitive data information present in that file. In that case, credentials.
+
+2. If you modify the file in your workspace, then commit and push it, will it be enough to erase this password information from the repository? (It's not). Why?
+
+> No. The sensitive information still present in the history and could be retrieved from there.
+
+3. If you delete the file and push it, then create a new one with the rest of the information, is it enough? (It's not). Why?
+
+> No. It's a different approach with the same result as above. The data still in the history.
+
+4. How to fix this? How do you remove something from Git history when it is in the remote repository?
+
+> It's necessary to rewrite the history and (force) push to remove the entries from the remote history.
+
+5. Which commands would you use? Explain what the command does.
+
+```
+# Using `filter-branch`
+
+To completely remove the entire file from the history
+
+```
+$ git filter-branch --force --index-filter \
```

Comment:
`git filter-repo` is the new tool recommended for the job by git-scm. Simpler to use and safer 👍

Reply:
I see, however, do you agree it doesn't mean that won't work?!

Reply:
I see. You're right. (https://github.com/newren/git-filter-repo)

However, do you agree it doesn't mean that won't work?!

Reply:
```
git filter-repo --path "my_env.txt" --invert-paths
git push --force
```

Reply:
filter-branch is still a valid answer.